addons/sale_timesheet/models/sale_order.py

```diff
@@ -66,8 +66,13 @@ def action_confirm(self):
                             order._create_analytic_account(prefix=line.product_id.default_code or None)
                         order.project_id.project_create({'name': order.project_id.name})
                         break
+            order.order_line.filtered(self._is_task)._create_task()
         return result
 
+    def _is_task(self, line):
```

Comment:
This can be decorated with `@api.model` (although I don't know if you have already deprecated this). Isn't also better to put this method at sale.order.line level with `self.ensure_one()` and removing `line` argument in favor of `self` (although it changes the `filtered` expression)?